### ISSUE TITLE
Problems Fine-Tuning Flan-T5 with Accelerate

### ISSUE THREAD

**Issue by experimarketing** [huggingface/accelerate#1006]
I am trying to fine tune flan-t5 with multi-gpu and accelerate, and the code seems to freeze at this step: 

`                loss = model(b_input_ids, attention_mask=b_input_mask, labels=lm_labels).loss
`

When it is frozen at this step, the GPU's are showing 100% usage and the memory usage is the same for each GPU. 

I ran `accelerate config` before starting. 
The settings I chose were the following: 
**Do you wish to use torch dynamo?** No
**deepspeed** : no
**fsdp**: yes
**shard**: full shard
**offload params and grads to cpu**: no
**wrap**: transformer based wrap
**transformer layer class name**: T5block
**prefetch policy**: backward pre
**dict**: sharded_state_dict
**gpus**: 8
**bf16 or fp16**: no

Then I ran `accelerate launch main.py` with no extra args.

When running I got two warnings: 
`FSDP Warning: When using FSDP, it is efficient and recommended to call prepare for the model before creating the optimizer

FSDP Warning: When using FSDP, several parameter groups will be conflated into a single one due to nested module wrapping and parameter flattening.`

This is part of the setup code: 
```
    model = T5ForConditionalGeneration.from_pretrained(model_version)
    tokenizer = AutoTokenizer.from_pretrained(model_version)
    accelerator = Accelerator(gradient_accumulation_steps=accumulation)
    device = accelerator.device
    model.to(device)
    optimizer = t5_AdamW_grouped_LLRD(model, learning_rate)
    contexts, labels = load_data(training_data)
    training_dataloader = create_dataloader(contexts, labels, batch_size, tokenizer)
    contexts, labels = load_data(evaluation_data)
    testing_dataloader = create_dataloader(contexts, labels, batch_size, tokenizer)
    model, optimizer, training_dataloader, testing_dataloader = accelerator.prepare(model, optimizer, training_dataloader, testing_dataloader)
    model, optimizer, train_losses = train_on_text(
        model,
        training_dataloader,
        accelerator,
        optimizer,
        tokenizer,
        train_losses,
        epochs=epochs_before_lr_adjustment,
    )
```

And this is the train_on_text function, up until it stops working: 
```
def train_on_text(model, dataloader, accelerator, optimizer, tokenizer, train_losses, epochs):
    train_loss_set = []
    model.train()
    for epoch_number in range(epochs):       
        tr_loss = 0  
        nb_tr_examples, nb_tr_steps = 0, 0
        step = 0
        for context, replies in dataloader:
            with accelerator.accumulate(model):
                step = step + 1
                if step % 1000 == 0:
                    if accelerator.is_main_process:
                        print("Step: ", step, flush=True)
                        print("Total time:", time.time() - start_time, flush=True)
                b_input_ids = context['input_ids']  
                b_input_mask = context['attention_mask']
                b_decoder_in = replies['input_ids']

                lm_labels = b_decoder_in[:, :].clone().detach()
                lm_labels[b_decoder_in[:, :] == tokenizer.pad_token_id] = -100
               # It doesn't make it past this next line:
                loss = model(b_input_ids, attention_mask=b_input_mask, labels=lm_labels).loss

```

This warning comes at the end as well: 
`/opt/conda/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py:2387: UserWarning: torch.distributed._all_gather_base is a private function and will be deprecated. Please use torch.distributed.all_gather_into_tensor instead.`

I have waited for up to 10 minutes for it to finish that one step, but it doesn't. And it still stays at 100% GPU usage.

**Comment by sgugger** [comment]
cc @pacman100

**Comment by experimarketing** [comment]
I ran this same code again but with a different accelerate config to see if I could get it working. 
```
In which compute environment are you running? This machine
Which type of machine are you using? multi-GPU
How many different machines will you use (use more than 1 for multi-node training)? [1]: 1
Do you wish to optimize your script with torch dynamo?[yes/NO]:no
Do you want to use DeepSpeed? [yes/NO]: no
Do you want to use FullyShardedDataParallel? [yes/NO]: no
Do you want to use Megatron-LM ? [yes/NO]: no
How many GPU(s) should be used for distributed training? [1]:8
What GPU(s) (by id) should be used for training on this machine as a comma-seperated list? [all]:0,1,2,3,4,5,6,7
Do you wish to use FP16 or BF16 (mixed precision)?no                   
```

This time I let it run for 20+ minutes, but it stopped at the `accelerator.prepare()` function this time.

For reference, the small training I am testing can run in less than 20 minutes on a single GPU.

**Comment by pacman100** [comment]
Hello @experimarketing, so as i understand, even the multi-gpu setup si stuck indefinitely. Could you please share a minimal reproducible script/example on our side to reproduce the issue. This would help us in deep dive and resolution.
